### PR TITLE
Get access token from refresh token

### PR DESCRIPTION
**Pull request by mores**
https://github.com/pac4j/pac4j/issues/1043

### CODE REVIEW

File: pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfileCreator.java [hunk 1]
```diff
@@ -13,12 +15,19 @@
  * @author Jerome Leleu
  * @since 1.9.2
  */
-public class AzureAdProfileCreator extends OidcProfileCreator<AzureAdProfile> {
+public class AzureAdProfileCreator extends OidcProfileCreator<AzureAdProfile,AzureAdOidcConfiguration> {
```
Comment (leleuj) [inline pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfileCreator.java]: In fact, we don't care about the configuration type, do we really need this change?

**Comment by leleuj** [comment]
Looks good to me, just a minor comment.

Can you update the doucmentation as well? -> https://github.com/pac4j/pac4j/blob/master/documentation/docs/clients/openid-connect.md

I remember a concern when updating the OIDC support for CAS: @mmoayyed : are you in line with this PR?

File: pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfileCreator.java (hunk 1 above)
Comment (mores) [reply pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfileCreator.java]: I can make it work without this change, but then I would store the tenant in the OidcConfiguration not in the AzureAdOidcConfiguration. I don't think that is what you want.

Can you suggest a way to make this work without this change ?

File: pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfile.java [hunk 2]
```diff
@@ -46,4 +66,33 @@ public String getUsername() {
         return (String) getAttribute(AzureAdProfileDefinition.UPN);
     }
 
+    public String getAccessTokenFromRefreshToken() {
```
Comment (mores) [inline pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfile.java]: Currently the String that is returned is json with attributes of:
token_type, scope, expires_in, ext_expires_in, expires_on, not_before, resource, access_token, refresh_token

Should it be left as is, or return some sort of AzureAdResponse ?

**Comment by mmoayyed** [comment]
Reviewed. Looks fine to me.

File: pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfile.java (hunk 2 above)
Comment (leleuj) [reply pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfile.java]: In fact, I missed the point here but this method should be in the `AzureAdClient`, taking an `AzureAdProfile` as input and returning only an access token.

File: pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfileCreator.java (hunk 1 above)
Comment (leleuj) [reply pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfileCreator.java]: For sure, we need the `AzureAdOidcConfiguration` to hold the tenant.

My point is: should we have the `OidcProfileCreator` parametrized with `? extends OidcConfiguration`?

File: pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfile.java (hunk 2 above)
Comment (mores) [reply pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfile.java]: How should the end user get a reference to the AzureAdClient ?

Comment (leleuj) [reply pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfile.java]: The `AzureAdClient` has been defined in the `Config`  component, so `config.findClient("AzureAdClient")` will do the job...

Comment (mores) [reply pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfile.java]: I don't think that is 100% correct....Here is the (verbose) gist as I see it:

`
org.pac4j.core.config.Config config = org.pac4j.core.config.ConfigSingleton.getConfig();
                org.pac4j.core.client.Clients clients = config.getClients();
                org.pac4j.oidc.client.AzureAdClient azureAdClient = clients.findClient( org.pac4j.oidc.client.AzureAdClient.class );
String accessToken = azureAdClient.getAccessTokenFromRefreshToken( azureAdProfile );
`

Or the one liner gist if you prefer:

`String accessToken = org.pac4j.core.config.ConfigSingleton.getConfig().getClients().findClient( org.pac4j.oidc.client.AzureAdClient.class ).getAccessTokenFromRefreshToken( azureAdProfile );`

If you approve this gist I will open up a second Pull Request.

Comment (leleuj) [reply pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfile.java]: In fact, you don't need to open a second PR: just move the `getAccessTokenFromRefreshToken` method from the `AzureAdProfile` to the `AzureAdClient`. Thanks

**Pull request closed without merging**